### DISCUSSION TITLE
Duda del schema generado

### DISCUSSION THREAD

**Discussion by francorojo** [dds-jv/foro#104] [Consultas de herramientas] [answered]
Buen diaaa, tenia una consulta, ademas de crear las tablas al final me agrego esto y no se si esta bien (y que significa en realidad tampoco muy bien, busque que era como una restriccion a ese atributo creo), agrego imagen:
![image](https://user-images.githubusercontent.com/49093389/92950243-bf1b4900-f432-11ea-9697-94743ff69c00.png)

Todas las demas tablas parecen bien la verdad, creo q igual a como las hubiera hecho directo en sql. Por las dudas agrego como quedo esa tabla generada con el create y ademas como me quedo en el codigo, porque tuve una particularidad:
![image](https://user-images.githubusercontent.com/49093389/92950275-c7738400-f432-11ea-9380-30ecce9dd3d5.png)

No me agarraba solo con poner el @id, me pedia un id especial y no me parecia que en este caso hacia falta uno especial aparte teniendo ya naturalmente la identificacion por las dos claves FK, entonces busque en internet y decia que implemente  serializable:

![image](https://user-images.githubusercontent.com/49093389/92950672-7021e380-f433-11ea-96c0-5600563067a0.png)

Gracias, saludo grande!

**Comment by francorojo** [comment]
La parte de arriba de la consulta ya me di cuenta al mirar de nuevo la clase :)
Si me gustaria saber por las dudas si esta bien implementar Serializable para poder poner los dos id.

**Comment by gastonprieto** [comment]
Buenas @francorojo !

No es necesario el Serializable, eso es para algunos contextos que no vamos a usar. Te dejo este post de StackOverflow que lo explica en detalle: https://stackoverflow.com/questions/2726300/do-hibernate-table-classes-need-to-be-serializable

Buenisimo que lo encontraste lo otro 🥇 
Si queres comentanos que era por si alguien lo ve en el futuro 👍 

Además acordate de pegarlo como texto las cosas así si alguien tire en el search el nombre de la tabla o algo parecido lo encuentra 👼

**Comment by francorojo** [comment, marked as answer]
Holaa, aaa buenísimo gracias! Si de una, espero pegarle xd, por lo que entendí era la forma en que se generaban las relaciones de las FK en las tablas, ese código raro es para identificar esa FK y generar la relación con la tabla a la cual le interesa la información. Lo que en código ponemos directamente por ejemplo, Jugador capitán; en realidad en la tabla se guarda la referencia a la PK del Jugador. (Sino habría un montón de información repetida si guardamos todos los datos de vuelta ademas de en la tabla Jugadores, en otra tabla que quiera referenciar a un jugador)
Ah dale, no pensé en ponerlo como texto por ese motivo, lo voy a tener en cuenta!
Saludooos!!!
